### ISSUE TITLE
GGMax - Bug - Animals need animations for turning

### ISSUE THREAD

**Issue by Kravenwolf** [TheGameCreators/GameGuruRepo#3252]
Currently when you "spook" a rabbit, skunk, etc., they are rotated to a different direction and then start running off, however, there doesn't appear to be an actual animation for them turning so they just sort of instantly teleport to a new direction and then run off. It looks a bit odd.

I assume this wouldn't be a priority, but maybe adding some additional frames for the animals so when they turn/rotate left or right, it looks more smooth and natural? (Not sure if fish have the same issue)

**Comment by AmenMoses** [comment]
So the problem here is with animations, i.e. if you want to mimic natural behaviour using animations then you need to have a huge number of base animations and in addition for each pair of animations a transition animation to smoothly go from one to the next.

Doing this would allow for extremely naturalistic behaviour but the logic required to make it all work becomes enormous and in reality the human eye is not going to really appreciate it anyhow.

Animations are used because they are efficient but there will always be a point where they just have to be 'good enough', granted a few more transitions would make for smoother visuals but is it really worth all the effort?

**Comment by Kravenwolf** [comment]
@AmenMoses, fair point. What about adding a quick rotation to the animals via scripting in the direction they're going to point to before they start running off in that direction (~<300-400ms)?

**Comment by AmenMoses** [comment]
They are using the built in navmesh navigation system (at least the mammals are) which is supposed to rotate them on the move but doesn't appear to be very smooth.  I tried different turn speeds but generally all that seems to achieve is that they run sidewise for a while which looks even worse.

Btw I recently fired up an U4 based game that I knew to contain various animals and the behaviour of the animals in that was virtually identical to the behaviour in MAX, i.e. the animals turn almost instantly so it is a common issue it seems.
